rust-runtime/aws-smithy-http-server/src/request/request_id.rs

```diff
@@ -201,11 +201,9 @@ where
     }
 
     fn call(&mut self, req: http::Request<Body>) -> Self::Future {
-        let request_id = req.extensions().get::<ServerRequestId>();
-        if request_id.is_none() {
-            return Box::pin(self.inner.call(req));
-        }
-        let request_id = request_id.unwrap().to_owned();
+        let request_id = req.extensions().get::<ServerRequestId>()
+            .expect("You must add `ServerRequestIdProviderLayer` to provide a request ID for the response. \
```

Comment:
Let's mention ordering here as well. They might have the layer but in the wrong position.